src/transformers/trainer.py

```diff
@@ -695,7 +695,10 @@ def _move_model_to_device(self, model, device):
     def _set_signature_columns_if_needed(self):
         if self._signature_columns is None:
             # Inspect model forward signature to keep only the arguments it accepts.
-            signature = inspect.signature(self.model.forward)
+            model_to_inspect = self.model
+            if is_peft_available() and isinstance(self.model, PeftModel):
+                model_to_inspect = self.model.base_model.model
```

Comment:
this would fail with Prompt Learning methods which don't have the `base_model.model` attribute.